File: Python/services/create_insert.py
from typing import List

from utils.crud_operations import CrudOperations
from models.note import Note


def create_insert(list_of_notes: List[Note]) -> None:
    # URI and database/collection names
    uri = "mongodb://localhost:27017"
    dbName = "noteTakingApp"
    collectionName = "notes"

    # Connect to MongoDB client
    client = CrudOperations.connect_to_client(uri)

    # Drop the database - before starting
    CrudOperations.drop_database(client, dbName)

    # Get the collection
    collection = CrudOperations.get_collection(client, dbName, collectionName)

    # Insert documents into the collection
    for note in list_of_notes:
        document = {
            "title": note.title,
            "content": note.content,
            "tags": note.tags,
            "created_on": note.created_on,
            "last_updated_on": note.last_updated_on
        }
        CrudOperations.insert_document(collection, document)

    # Close the MongoDB client connection
    CrudOperations.close_client(client)

    # Temp: Make the output file empty
    outputFilePath = "output.txt"
    try:
        with open(outputFilePath, 'w') as f:
            f.write('')
    except Exception as e:
        print("Error writing output file: ", e)

File: Go/services/create_insert.go
package services

import (
	"os"

	"github.com/siddhant-vij/Note-Taking-App/models"
	"github.com/siddhant-vij/Note-Taking-App/utils"
)

const (
	dbName         = "noteTakingApp"
	collectionName = "notes"
)

func CreateInsert(uri string, notes []models.Note) {
	// Connect to MongoDB client
	client, _ := utils.ConnectToClient(uri)

	// Drop the database - before starting
	utils.DropDatabase(client, dbName)

	// Get the collection
	collection := utils.GetCollection(client, dbName, collectionName)

	// Insert documents into the collection
	for _, note := range notes {
		utils.InsertDocument(collection, note)
	}

	// Close the MongoDB client connection
	utils.CloseClient(client)

	// Temp: Make the output file empty
	outputFile, _ := os.Create("output.txt")
	outputFile.Close()
}
